lib/app.rb

```diff
@@ -0,0 +1,92 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'classroom'
+require_relative 'rental'
+require_relative 'person_creator'
+require_relative 'book_creator'
+require_relative 'rental_creator'
+
+# App class for managing rentals of rentals of the books to people
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each do |book|
+      puts "Title:\"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    @people.each do |person|
+      puts "[#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person()
+    puts 'Do you want to create a Student (1) or Teacher (2)? [Input the number]:'
+    choice = gets.chomp.to_i
+    person_creator = PersonCreator.new(@people)
+    case choice
+    when 1
+      person_creator.create_student
+    when 2
+      person_creator.create_teacher
+    end
+  end
+
+  def create_book()
+    book_creator = BookCreator.new(@books)
+    book_creator.create_book
+  end
+
+  def create_rental()
+    rental_creator = RentalCreator.new(@books, @people, @rentals)
+    rental_creator.create_rental
+  end
+
+  def list_rentals_by_person_id
+    person_id = read_person_id_from_user_input
+    person = find_person_by_id(person_id)
+
+    return unless person
+
+    rentals = get_rentals_by_person(person)
+
+    if rentals.empty?
+      puts "#{person.name} (id: #{person.id}) has no rentals."
+    else
+      puts "All rentals for #{person.name} (id: #{person.id}):"
+      print_rentals(rentals)
+    end
+  end
```

Comment:
Overall, your `list_rentals_by_person_id` method looks good and effectively lists the rentals for a given person. However, I have a suggestion to improve the user experience.

**Suggestion:**
To enhance user convenience, consider modifying the method to also list the names and IDs of all available people. This way, users can quickly select the desired person by their name or ID without having to look up the ID separately.

**Implementation:**
1. Before reading the person ID from user input, retrieve and display a list of all available people with their names and IDs. You can create a separate method to fetch and print this list.
2. After displaying the list of people, prompt the user to enter the person's ID for whom they want to view the rentals.
3. Proceed with the rest of your existing logic to list their rentals.

This modification will make your application more user-friendly and efficient, as users won't have to remember or look up the IDs separately when trying to view rentals for a specific person.

Feel free to reach out if you have any questions or need further assistance. Keep up the good work!